S24-testing/15-done-testing.t

```diff
@@ -39,16 +44,25 @@ END_PROG_1
 END_PROG_2
 
 my @test =
-    { expected => False, description => 'wrong count, all pass' },
-    { expected => True,  description => 'correct count, all pass' },
-    { expected => False, description => 'correct count, one fails' },;
+    {   description => 'wrong count, all pass',
+        expected => {
+            out => sub { 'False' eq ( $^a ~~ rx/ \w+ \s+ $ / ).trim; },
```

Comment:
Why is anchoring to the end of the string not sufficient?

Reply:
Because bugs are crafty devils. There is no warranty that the last line comes from _your_ code and not a result of some complicated set of events causing other code in compiler to also print some boolean. For example, think of a accidentally left behind debug output which your code provokes to get activated. In this case it could mask a problem your test is supposed to catch.

By putting what you expect to see into the context you actually lower the probability of such breakage to almost negligible value.

Reply:
I agree with all you say.  I differ by seeing the problem as negligible to start.  Not so much in my code having the problem you describe, but that it would matter.  `use Test; plan $n; ok  True; ... done-testing;`  is such a common use of `Test`, that it is hard to claim as my code.

But I comfortably  bow to your authority as my small experience has been with much smaller projects.

Reply:
Please, get me correct: I don't need bowing, just sharing experience.

The point of saying "your code" is to distinguish the code coming from your side of the test from the code on the compiler side. The test snippets (_programs_ in the terms of your test) are your code because it's what you test. And the rest of what I said above is related to the fact that nothing has to be accepted as granted. The output you get from your snippets could be not only the output of `Test` routines and your final `say`. It may come from some buggy code in Rakudo itself, for example. So, by putting what you test into context you apply additional layer of protection from such unwanted interventions from the core code. However unlikely such interventions are.

Anyway, I saw the last commit and things looks really good now. It's too late to review it carefully today. Will do it tomorrow.

Thank you!